components/camel-kafka/src/main/java/org/apache/camel/component/kafka/consumer/support/KafkaRecordProcessor.java

```diff
@@ -149,14 +146,30 @@ private boolean processException(
                 Exception exc = exchange.getException();
                 LOG.warn("Error during processing {} from topic: {} due to {}", exchange, topicPartition.topic(),
                         exc.getMessage());
-                LOG.warn("Will seek consumer to offset {} on partition {} and start polling again.",
-                        lastResult.getPartitionLastOffset(), lastResult.getPartition());
+                LOG.warn("Will seek consumer to offset {} on partition {} and start polling again.", 
+                        record.offset(), record.partition());
             }
 
             // force commit, so we resume on next poll where we failed 
             // except when the failure happened at the first message in a poll
             if (lastResult.getPartitionLastOffset() != AbstractCommitManager.START_OFFSET) {
-                commitManager.forceCommit(topicPartition, lastResult.getPartitionLastOffset());
+                // the record we are processing had the error 
+                // so we will force commit the offset prior 
+                // this will enable the current desired behavior to 
+                // retry the message 1 more time
+                //
+                // Note: without a more extensive look at handling of breakOnFirstError
+                // we will still need the lastResult so that we don't force 
+                // retrying this message over and over
+                // commitManager.forceCommit(topicPartition, record.offset() - 1);
+                
+                // we should just do a commit (vs the original forceCommit)
+                // when route uses NOOP Commit Manager it will rely
+                // on the route implementation to explicitly commit offset
+                // when route uses Synch/Asynch Commit Manager it will 
+                // ALWAYS commit the offset for the failing record
+                // and will ALWAYS retry it
+                commitManager.commit(topicPartition);
             }
```

Comment:
This is the key change in behavior
It goes beyond a fix and gives route implementers ability to choose how breakOnFirstError should work depending on the commit manager they use

Reply:
> This is the key change in behavior It goes beyond a fix and gives route implementers ability to choose how breakOnFirstError should work depending on the commit manager they use

Thanks for the heads up. 

I am OK with this given the scope of the change. The important thing here is that it needs to be documented on both the component doc as well as on the upgrade guide.

Reply:
Happy to do that. Can you point me to where to update those

Reply:
The following docs need to be adjusted:

* [The Camel 3.21 upgrade guide](https://github.com/apache/camel/blob/main/docs/user-manual/modules/ROOT/pages/camel-3x-upgrade-guide-3_21.adoc)
* The [`breakOnFirstError`](https://github.com/apache/camel/blob/main/components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConfiguration.java#L856-L866) documentation.
*  

On the the 3.21 upgrade guide is OK to be very brief: just saying that the behavior for the `breakOnFirstError` has changed and how, and then tell that the users should consult the component documentation. 

You can complement the details about how it was changed, how it behaves, what to expect, etc on the `breakOnFirstError` documentation. If necessary, you can also add a section on the [The Kafka Component doc](https://github.com/apache/camel/blob/main/components/camel-kafka/src/main/docs/kafka-component.adoc).

Reply:
Updated docs here
https://github.com/apache/camel/pull/11970

Reply:
Thanks for adding the doc changes to this PR.